tools/clang-display-context/ClangDispCont.cpp

```diff
@@ -0,0 +1,169 @@
+//===- ClangDispCont.cpp ------------------------------------------------===//
+//
+//                     The LLVM Compiler Infrastructure
+//
+// This file is distributed under the University of Illinois Open Source
+// License. See LICENSE.TXT for details.
+//
+//===--------------------------------------------------------------------===//
+//
+// Clang tool which generates and outputs an issue string based on SA,
+// from the location (line, column number) of the issue.
+//
+//===--------------------------------------------------------------------===//
+
+#include "clang/AST/ASTConsumer.h"
+#include "clang/AST/ASTContext.h"
+#include "clang/AST/DeclTemplate.h"
+#include "clang/AST/ExprCXX.h"
+#include "clang/Basic/SourceManager.h"
+#include "clang/Frontend/CompilerInstance.h"
+#include "clang/Frontend/FrontendActions.h"
+#include "clang/StaticAnalyzer/Core/IssueHash.h"
+#include "clang/Tooling/CommonOptionsParser.h"
+#include "clang/Tooling/Tooling.h"
+#include "llvm/ADT/Twine.h"
+#include "llvm/Support/CommandLine.h"
+#include "llvm/Support/Signals.h"
+
+#include <regex>
+#include <string>
+
+using namespace llvm;
+using namespace clang;
+using namespace clang::tooling;
+
+static cl::OptionCategory ClangDispContCategory("clang-disp-context options");
+
+static cl::opt<uint> Line(
+    "line",
+    cl::desc("Line number of the issue's position. \n"),
+    cl::cat(ClangDispContCategory));
+
+static cl::opt<uint> Col(
+    "col",
+    cl::desc("Column number of the issue's position. \n"),
+    cl::cat(ClangDispContCategory));
+
+class DisplayContextConsumer : public ASTConsumer {
+public:
+  DisplayContextConsumer(ASTContext &Context) : Ctx(Context) {}
+
+  ~DisplayContextConsumer() {
+    // Flush results to standard output.
+    outs() << IssueString;
+  }
+
+  virtual void HandleTranslationUnit(ASTContext &Ctx) {
+    if (!Ctx.getDiagnostics().hasErrorOccurred())
+      handleDecl(Ctx.getTranslationUnitDecl());
+  }
+
+private:
+  template <typename T> 
+  bool checkIfInteresting(const T *N);
+  void handleDecl(const Decl *D);
+  void handleStmt(const Stmt *S);
+  void setEnclosingDecl(const Decl *D);
+  void setIssueString(const std::string &S);
+  
+  ASTContext &Ctx;
+  FullSourceLoc IssueLoc;
+  const Decl *EnclosingDecl;
+  std::string IssueString;
+};
+
+template <typename T>
+bool DisplayContextConsumer::checkIfInteresting(const T *Node) {
+  // Nodes are only interesting if they contain the line with the issue
+  if (!Node)
+    return false;
+
+  FullSourceLoc L1 = Ctx.getFullLoc(Node->getLocStart());
+  FullSourceLoc L2 = Ctx.getFullLoc(Node->getLocEnd());
```

Comment:
Not that `LocEnd` is the beginning of the last token associated with the node.
If you want the real end position you need to account for the token length. 
You could use `getLocForEndOfToken` of Lexer for example to circumvent this problem.